arviz/data/inference_data.py

```diff
@@ -589,6 +586,108 @@ def to_dataframe(
             (dfs,) = dfs.values()
         return dfs
 
+    def to_zarr(self, store=None):
+        """
+        Convert InferenceData to a :class:`zarr.hierarchy.group` using the same groups names
+        as in InferenceData.
+
+        Data groups ("observed_data", "constant_data", "predictions_constant_data") are
+        skipped.
+
+        Raises TypeError if no valid store is found.
+
+        Parameters
+        ----------
+        store: zarr.storage i.e MutableMapping or str, optional
+            Zarr storage class or path to desired DirectoryStore.
+
+        overwrite: bool, optional,
+            If True, delete any pre-existing data in store at path before creating the groups.
+
+        Returns
+        -------
+        zarr.hierarchy.group
+            A zarr hierarchy group containing the InferenceData.
+
+        References
+        ----------
+        https://zarr.readthedocs.io/
+        """
+
+        import zarr
+        from collections.abc import MutableMapping
+
+        # Check store type and create store if necessary
+        if store is None:
+            store = zarr.storage.TempStore(suffix="arviz")
+        elif isinstance(store, str):
+            store = zarr.storage.DirectoryStore(path=store)
+        elif not isinstance(store, MutableMapping):
+            raise TypeError(f"No valid store found: {store}")
+
+        groups = self.groups()
+        if "observed_data" in groups:
+            groups.remove("observed_data")
+        if "constant_data" in groups:
+            groups.remove("constant_data")
+        if "predictions_constant_data" in groups:
+            groups.remove("predictions_constant_data")
+
+        if len(groups) == 0:
+            raise TypeError("No valid groups found!")
+
+        for group in groups:
+            # Create zarr group in store with same group name
+            getattr(self, group).to_zarr(store=store, group=group, mode="w")
+
+        return zarr.open(store)  # Open store to get overarching group
```

Comment:
How does zarr handle open files ?

Are they closed manually?

Reply:
As far as I understand it only keeps metafiles to the location of the files and loads them on demand.

From the [documentation](https://zarr.readthedocs.io/en/stable/_modules/zarr/storage.html?highlight=close#):
Files are only held open while they are being read or written and are
closed immediately afterwards, so there is no need to manually close any files.

FYI: There is a great talk from one of the developers [here](https://www.youtube.com/watch?time_continue=2&v=qyJXBlrdzBs&feature=emb_title).